frontend/src/Calendar/Agenda/Agenda.js

```diff
@@ -1,14 +1,50 @@
 import moment from 'moment';
 import PropTypes from 'prop-types';
 import React from 'react';
+import minNumberInArray from 'Utilities/Number/minNumberInArray';
 import AgendaEventConnector from './AgendaEventConnector';
 import styles from './Agenda.css';
 
 function Agenda(props) {
   const {
-    items
+    items,
+    start,
+    end
   } = props;
 
+  const startDateParsed = Date.parse(start);
+  const endDateParsed = Date.parse(end);
+
+  if (items) {
+    for (const i in items) {
+      if (items[i]) {
```

Comment:
Seems like alot of nested logic, can just use `forEach` which is used everywhere else in UI?
```
items.forEach((item) => {
  const cinemaDateParsed = Date.parse(item.inCinemas);
  const digitalDateParsed = Date.parse(item.digitalRelease);
  const physicalDateParsed = Date.parse(item.physicalRelease);
  const dates = [];
  
  if (cinemaDateParsed > 0 && cinemaDateParsed >= startDateParsed && cinemaDateParsed <= endDateParsed) {
    dates.push(cinemaDateParsed);
  }
  if (digitalDateParsed > 0 && digitalDateParsed >= startDateParsed && digitalDateParsed <= endDateParsed) {
    dates.push(digitalDateParsed);
  }
  if (physicalDateParsed > 0 && physicalDateParsed >= startDateParsed && physicalDateParsed <= endDateParsed) {
    dates.push(physicalDateParsed);
  }

  item.sortDate = Math.min(...dates);
  item.cinemaDateParsed = cinemaDateParsed;
  item.digitalDateParsed = digitalDateParsed;
  item.physicalDateParsed = physicalDateParsed;
}); 

items.sort((a, b) => ((a.sortDate > b.sortDate) ? 1 : -1));
```